bentoml/_internal/server/http_app.py

```diff
@@ -272,17 +274,32 @@ def on_startup(self) -> list[t.Callable[[], None]]:
 
     async def readyz(self, _: "Request") -> "Response":
 
-        if all(
-            await asyncio.gather(
-                *(
-                    runner.runner_handle_is_ready()
-                    for runner in self.bento_service.runners
-                )
-            )
-        ):
-            return PlainTextResponse("\n", status_code=200)
+        runner_timeout = BentoMLContainer.runners_config.timeout.get()
+        check_interval = 5
+        start_time = time.time()
+        logger.debug(
+            "Waiting for runners {!r} to be ready...".format(self.bento_service.runners)
+        )
+        while time.time() - start_time < runner_timeout:
+            try:
+                if all(
+                    await asyncio.gather(
+                        *(
+                            runner.runner_handle_is_ready()
+                            for runner in self.bento_service.runners
+                        )
+                    )
+                ):
+                    return PlainTextResponse("\n", status_code=200)
+                else:
+                    time.sleep(check_interval)
+            except (ConnectionError, socket.timeout, HTTPException) as e:
+                logger.debug("[%s] Retrying ..." % e)
+                time.sleep(check_interval)
 
-        raise HTTPException(500)
+        raise RuntimeError(
+            f"Timed out waiting {runner_timeout} seconds for runners to be ready."
+        )
```

Comment:
I think we can return HTTPException here instead of raising RuntimeError

Reply:
Note that this code diff is outdated. As of commit `26a349b`, I added `HTTPException` to the try-catch block's exceptions -- Let me know if that looks better. 

My understanding was that we should leave `RuntimeError` in here in case the runners take too long for some reason. Could you clarify for me why we should remove it?